site-src/geps/gep-1364.md

```diff
@@ -0,0 +1,409 @@
+to reporting "something went wrong" than they are "everything's okay".
+
+Not sure if pro or con:
+* Leans the design towards favoring conditions always being present, since you
+can't be sure if everything is good unless you see `AllGood: true`. The absence
+of a positive-polarity condition implies that the condition could be false. This
+puts this option more in line with the API guidelines on this point.
+
+#### Some Conditions Positive
+
+In this case, only a limited set of summary conditions are positive, and the rest
+are negative.
+
+Pros:
+* Error states can be described with `Error: true` instead of `NoError: false`.
+* Negative polarity error conditions are more friendly to not being present (since
+absence of `Error: true` implies everything's okay).
+
+Cons:
+* Any code handling conditions will need a list of the positive ones, and will
+need to assume that any others are negative.
+
+#### Decision
+
+Gateway API conditions will be positive for conditions that describe the happy
+state of the object, which is currently `Attached` and `ResolvedRefs`, and will 
+also include the new `Programmed` condition, and the newly-Extended condition
+`Ready`.
+
+
+### Should conditions always be added?
+
+Not all of them.
+
+Positive polarity Conditions that describe the desirable state of the object must
+always be set. These are currently `Attached`, `ResolvedRefs`, and `Programmed`.
+Implementations that use `Ready` must also add it immediately, set to `false`.
+
+### New and Updated Conditions
+
+#### `Attached`
+
+This GEP proposes replacing all conditions that indicate syntactic and semantic
+validity with one, `Attached` condition type, with the exception of the
+GatewayClass resource (because it's the root of the resource tree and doesn't
+attach to anything).
+
+That is, the proposal is to replace:
+* `Scheduled` on Gateway
+* `Detached` on Listener
+* `Accepted` on Route
+
+with `Attached` in all these locations.
+
+GatewayClass will maintain the `Accepted` condition.
+
+All of these conditions share the following meanings:
+* The resource has been accepted for processing by the controller
+* The resource is syntatically and semantically valid, and internally consistent
+* The resource fits into a larger system of Gateway API resources, and there is
+is no missing information, including but not limited to:
+  * Any mandatory references resolve to existing resources (examples here are the
+  Gateway's gatewayClass field, or the `parentRefs` field in Route resources)
+  * Any specified TLS secrets exist
+* The resource is supported by the controller by ensuring things like:
+  * Any Kinds being referred to by the resource are supported
+  * Features being used by the resource are supported
+
+All of these rules can be summarized into:
+* The resource is valid enough to produce some configuration in the underlying
+data plane.
+
+For Gateway, `Attached` also subsumes the functions of `Scheduled`: `Attached`
```

Comment:
On some impls, provisioning takes about a minute. That may mean that some errors that are immediately known are masked. Or I suppose they aren't masked, but we need to wait the entire minute to know if its valid. The message field could say "all fields valid, waiting for deployment" perhaps, but that isn't in the machine parse-able section.

That may not warrant change, just a thought

Reply:
I worded this carefully, so as not to imply that the provisioning needs to have _completed_, just that _it can be started_. The completion of infra provisioning should be handled by `Programmed` or `Ready`, or both.